activities.py

```diff
@@ -0,0 +1,62 @@
+# @@@SNIPSTART python-money-transfer-project-template-withdraw
+from temporalio import activity
+
+from shared import PaymentDetails
+from banking_service import BankingService, InvalidAccountError, InsufficientFundsError
+
+# Generic errors for activity failures
+class WithdrawFailedError(Exception):
+    pass
+
+class DepositFailedError(Exception):
+    pass
+
+class RefundFailedError(Exception):
+    pass
+
+@activity.defn
+async def withdraw(data: PaymentDetails) -> str:
+    reference_id: str = f"{data.reference_id}-withdrawal"
+    bank: BankingService = BankingService("bank-api.example.com")
+    try:
+        confirmation: str = bank.withdraw(data.source_account, data.amount, reference_id)
+        return confirmation
+    except InsufficientFundsError as error:
+        raise error
+    except InvalidAccountError as error:
+        raise error
+    except Exception as error:
+        raise WithdrawFailedError(error) from error
```

Comment:
If you're setting the error as the cause, don't usually need to also set it as the message